cockpit/src/subscriptions-view.jsx

```diff
@@ -27,11 +27,61 @@ import { InsightsStatus } from './insights.jsx';
 let _ = cockpit.gettext;
 
 class Listing extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            attaching_in_progress: false,
+            attaching_in_progress_promise: null,
+            attaching_message: _("Auto-attach")
+        };
+        this.handleAutoAttach = this.handleAutoAttach.bind(this);
+        this.updateProgress = this.updateProgress.bind(this);
+    }
+    updateProgress(msg) {
+        this.setState({ attaching_message: msg });
+    }
+    handleAutoAttach(event) {
+        // only consider primary mouse button
+        if (!event || event.button !== 0)
+            return;
+        if (this.props.autoAttach) {
+            let self = this;
+            this.setState({
+                attaching_in_progress: true,
+                attaching_message: _("Auto-attaching ...")
+            });
+            this.props.autoAttach()
+                .done(function () {
+                    self.setState({
+                        attaching_in_progress: false,
+                        attaching_message: _("Auto-attach")
+                    });
+                })
+                .fail(function () {
+                    self.setState({
+                        attaching_in_progress: false,
+                        attaching_message: _("Auto-attach")
+                    });
+                })
+                .progress((message) => {
+                    this.updateProgress(message);
+                });
+        }
+    }
     render() {
         if (this.props.children) {
+            let auto_attach_btn_disabled;
+            auto_attach_btn_disabled = this.state.attaching_in_progress || this.props.status === 'Unknown';
             return (
                 <div>
-                    <h2>{this.props.title}</h2>
+                    <div className="installed-products-line">
+                        <h2 className="installed-products-title">{this.props.title}</h2>
+                        <button className="btn btn-default auto-attach-btn"
```

Comment:
This button needs a test.

Reply:
Test added and fixed several other failed integration tests.

Reply:
Thanks!